### PR TITLE
Fix: Allow vespa indexing direct from s3

### PR DESCRIPTION
**Pull request by joel-wright**

### CODE REVIEW

File: src/index/vespa.py [hunk 1]
```diff
@@ -133,7 +135,11 @@ def get_document_generator(
 
     physical_document_count = 0
     for task in tasks:
-        embeddings = np.load(str(embedding_dir_as_path / f"{task.document_id}.npy"))
+        task_array_file_path = cast(Path, embedding_dir_as_path / f"{task.document_id}.npy")
+        task_array_file_like = task_array_file_path.open('rb')
+        task_array_file_like.seek(0)
+        embeddings = np.load(BytesIO(task_array_file_like.read()))
+        task_array_file_like = None
```
Comment (olaughter) [inline src/index/vespa.py:142]: What do you think about using a context manager for this? That way it gets closed and you'd get out of having to reset it for the next loop. Something like this:

```suggestion
        with open(task_array_file_path, "rb") as task_array_file_like:
            task_array_file_like.seek(0)
            embeddings = np.load(BytesIO(task_array_file_like.read()))
```

**Review by olaughter — APPROVED** [review]
Looks good. So this still preserves the ability to use a local path? 

One other suggestion below.

File: src/index/vespa.py (hunk 1 above)
Comment (joel-wright) [reply src/index/vespa.py:142]: the context manager is a good point, but I'll still need to remove the reference to the BytesIO - new revision coming

Comment (olaughter) [reply src/index/vespa.py:142]: 👍

Comment (joel-wright) [reply src/index/vespa.py:142]: no, your version is good :)

**Pull request merged**